integr8ly/backup-container-image
# Rbac

Adds the rbac resources to allow `oc` to be used inside the backup containers using a service account. Unfortunately adding those resources directly to the template does not work because creating the jobs would then fail if they already existed.

Verification steps:

1. run `oc apply -f templates/openshift/rbac`
1. Try using oc in your backup script
1. Create the job using for example:
```
oc new-app \
-f backup-job-template.yaml \
-p 'COMPONENT=mysql' \
-p 'COMPONENT_SECRET_NAME=sample-mysql-secret' \
-p 'BACKEND_SECRET_NAME=sample-s3-secret' \
-p 'ENCRYPTION_SECRET_NAME=sample-gpg-secret' \
-p 'IMAGE=<your dev image>' \
-p 'NAME=resources-backup' -n webapp
```
4. It should work

# Notes

If you need permissions to other resources than configmaps or secrets, feel free to add them to the clusterrole.
